src/components/StarRating/StarRating.jsx

```diff
@@ -7,40 +7,32 @@ class StarRating extends Component {
     super(props);
 
     this.state = {
-      stars: [...Array.from({ length: 5 }, (_, i) => i + 1)],
-      rating: 0,
-      hovered: 0
+      hoverAnimation: 0
     };
   }
-
-  changeRating(newRating) {
-    this.setState({ rating: newRating }, () => {
-      this.getStarRate(this.state.rating);
-    });
-  }
-
-  hoverRating(rating) {
-    this.setState({ hovered: rating });
-  }
-
   getStarRate(star) {
     this.props.onChange(star);
   }
 
+  hoverRating(star) {
+    this.setState({ hoverAnimation: star });
+  }
+
   render() {
-    const { stars, rating, hovered } = this.state;
+    const { stars, rating } = this.props.value;
```

Comment:
no need to pass an object in `value` prop, make it simplier :)

Reply:
Hope fixed in the right way :)))